### PR TITLE
Missing display labels should be treated as null

### PR DESCRIPTION
**Pull request by RohitPtnkr1996**
Fixes https://github.com/iTwin/imodel-native/issues/552

The parser should treat an unset override as there being no override and the default label should be shown.

Setting the override to an empty string should result in no label (the empty string) being shown.

itwinjs-core: https://github.com/iTwin/itwinjs-core/pull/6246

### CODE REVIEW

**Review by rschili — APPROVED** [review]

**Review by rschili — APPROVED** [review]

**Comment by ColinKerr** [comment]
I'm a bit concerned about treating empty string the same as null string.  Reviewing the code it looks like the original intention was to differentiate them with an empty string meaning no unit label is shown and null/no override meaning the original units display label is shown.

See test here where format override is explicitly tested to handle no label [M] and an empty label [M|] differently.
https://github.com/iTwin/imodel-native/blob/7ab960352e77e98fd0930356efb101a8500ac2cc/iModelCore/Units/tests/NonPublished/FormatTests.cpp#L369

The UnitProxy used in CompositeValueSpec uses an explicit flag to see if it has a label: https://github.com/iTwin/imodel-native/blob/7ab960352e77e98fd0930356efb101a8500ac2cc/iModelCore/Units/PublicAPI/Formatting/FormattingApi.h#L332

I'm not sure this ability is useful.  In a composite unit it allows you to show unit labels for some units and not for others.  For format overrides it allows you to turn off unit labels as part of the override.

I think the right thing to do is retain the differentiation and modify all code to handle it consistently.  

So when working with the array of units we should handle it like this: https://github.com/iTwin/imodel-native/blob/7ab960352e77e98fd0930356efb101a8500ac2cc/iModelCore/Units/src/Formatting/Format.cpp#L322

ECDb, JSON and Xml should persist empty string as empty string and unset as null.  Since the existing code will persist as empty string no matter what we may have flipped some bits in the past so we should add a test to ensure that a schema upgrade will cleanly set to null if unset in ECDb.  We should only rely on the value of the label during deserialization, when working with the object model we should rely on the flag telling us if the unit label is set.

As two follow on tasks I'd like to test the iTwin.js library to ensure it behaves correctly and update the documentation for iTwin.js (EC Conceptual docs) to clearly state the behavior in both the format override case and on format definitions.  File an issue for this against itwinjs-core and pick it up right after this so we can square the circle while it's all fresh.

File: iModelCore/ecobjects/src/Format.cpp [hunk 1]
```diff
@@ -488,18 +488,35 @@ SchemaWriteStatus ECFormat::WriteXml(BeXmlWriterR xmlWriter, ECVersion ecXmlVers
             {
             xmlWriter.WriteElementStart(FORMAT_COMPOSITE_UNIT_ELEMENT);
             if(label.IsValid())
-                xmlWriter.WriteAttribute(COMPOSITE_UNIT_LABEL_ATTRIBUTE, label.Value().c_str());
+                {
+                if (const auto labelValue = label.Value().c_str(); !Utf8String::IsNullOrEmpty(labelValue))
```
[the remark this answers is not in the document]
Comment (RohitPtnkr1996) [reply iModelCore/ecobjects/src/Format.cpp:492]: I was limiting the scope of the labelValue variable to just the if condition that requires it.
But yeah, I can pull out the assignment and flip the negation.

[the remark this answers is not in the document]
Comment (RohitPtnkr1996) [reply iModelCore/ecobjects/src/Format.cpp]: Done !

File: iModelCore/ECDb/Tests/NonPublished/SchemaUpgradeTests.cpp [hunk 2]
```diff
@@ -13542,15 +13542,17 @@ TEST_F(SchemaUpgradeTestFixture, Formats)
...
                                             <Unit label="m">MyMeter</Unit>
+                                            <Unit label="">u:CM</Unit>
                                             <Unit label="mm">u:MM</Unit>
                                         </Composite>
                                     </Format>
                                 </ECSchema>)xml")));
 
     assertFormat(m_ecdb, "MyFormat", "My Format", "",
                  JsonValue(R"json({"roundFactor":0.3, "type": "Fractional", "showSignOption": "OnlyNegative", "formatTraits": ["trailZeroes", "keepSingleZero"], "precision": 4, "decimalSeparator": ".", "thousandSeparator": ",", "uomSeparator": " "})json"),
-                 JsonValue(R"json({"includeZero":true, "units": [{"name":"MyMeter", "label":"m"}, {"name":"MM", "label":"mm"}]})json"));
```
Comment (ColinKerr) [inline iModelCore/ECDb/Tests/NonPublished/SchemaUpgradeTests.cpp:13553]: I think we should have a test that verifies we can go from empty string to unset and unset to empty string in a schema update.  Making KM empty string and CM unset in the upgraded schema would test that

Comment (RohitPtnkr1996) [reply iModelCore/ECDb/Tests/NonPublished/SchemaUpgradeTests.cpp:13553]: Done !

**Review by ColinKerr — APPROVED** [review]

**Review by rschili — APPROVED** [review]
